compiler/passes/cleanup.cpp

```diff
@@ -148,31 +149,34 @@ destructureTupleAssignment(CallExpr* call) {
   }
 }
 
-
 static void flatten_primary_methods(FnSymbol* fn) {
   if (TypeSymbol* ts = toTypeSymbol(fn->defPoint->parentSymbol)) {
     Expr* insertPoint = ts->defPoint;
+
     while (toTypeSymbol(insertPoint->parentSymbol))
       insertPoint = insertPoint->parentSymbol->defPoint;
+
     DefExpr* def = fn->defPoint;
+
     def->remove();
+
     insertPoint->insertBefore(def);
+
     if (fn->userString && fn->name != ts->name) {
       if (strncmp(fn->userString, "ref ", 4) == 0) {
         // fn->userString of "ref foo()"
         // Move "ref " before the type name so we end up with "ref Type.foo()"
         // instead of "Type.ref foo()"
-        fn->userString = astr("ref ", ts->name, ".", fn->userString+4);
+        fn->userString = astr("ref ", ts->name, ".", fn->userString + 4);
       } else {
         fn->userString = astr(ts->name, ".", fn->userString);
       }
     }
-    if (ts->hasFlag(FLAG_SYNC))
```

Comment:
Same question here. Isn't this no longer needed just because the current version happens to use secondary methods. Would this code be needed if they were primary methods?